src/installer/corehost/cli/hostpolicy/hostpolicy_context.cpp

```diff
@@ -158,11 +158,15 @@ int hostpolicy_context_t::initialize(hostpolicy_init_t &hostpolicy_init, const a
     fx_definition_vector_t::iterator fx_curr = fx_begin;
     while (fx_curr != fx_end)
     {
-        if (fx_curr != fx_begin)
+        if (fx_curr != fx_begin && app_context_deps_str != L"")
             app_context_deps_str += _X(';');
 
-        app_context_deps_str += (*fx_curr)->get_deps_file();
+        auto deps_str = (*fx_curr)->get_deps_file();
         ++fx_curr;
+        if (bundle::info_t::is_single_file_bundle() && (get_directory(deps_str) == args.app_root))
+            continue;
```

Comment:
This is potentially fragile.
Note that we already have a similar problem with libhost where we also don't want the "app" FX definition to show up in the list (since it's "fake" - there's no app in that case). This is handled inside the `get_app_fx_definition_range`.

I think a cleaner fix would be to:
* rename `deps_resolver_t::get_app_fx_definition_range` to something like `get_app_context_deps_files` to make it very specific (the API is already only used from this one spot anyway)
* Add the check into the method, so the condition would look like:
  ```C++
    if ((m_host_mode == host_mode_t::libhost || bundle::info_t::is_single_file_bundle())
        && begin_iter != end_iter)
  ```

And then the callsite can remain as-is.